Sources/NetworkProtection/Diagnostics/NetworkProtectionTunnelFailureMonitor.swift

```diff
@@ -52,7 +52,7 @@ public actor NetworkProtectionTunnelFailureMonitor {
         task?.isCancelled == false
     }
 
-    private weak var tunnelProvider: PacketTunnelProvider?
+    private let handshakeReporter: HandshakeReporting
```

Comment:
Tried with this to remove the dependency that was there. Made it so that the Wireguard adapter is accessed directly. Alternatively could have made the `PacketTunnelProvider` implement this and kept it `weak` in a mode delegate-like fashion.

Reply:
This looks like a good abstraction to me.